iris-mpc-common/src/test.rs

```diff
@@ -52,25 +70,80 @@ fn get_shared_template(
     is_valid: bool,
     template: &IrisCode,
     rng: &mut StdRng,
+    mirror_attack_testcase_hack: bool,
 ) -> (
     [GaloisRingIrisCodeShare; 3],
     [GaloisRingTrimmedMaskCodeShare; 3],
+    [GaloisRingIrisCodeShare; 3],
+    [GaloisRingTrimmedMaskCodeShare; 3],
 ) {
     let mut shared_code =
         GaloisRingIrisCodeShare::encode_iris_code(&template.code, &template.mask, rng);
 
-    let shared_mask = GaloisRingIrisCodeShare::encode_mask_code(&template.mask, rng);
+    // ugly hack to accomodate the mirror attack test case
+    // we need to simulate an incoming mirror attack iris code and mask
+    // generate_query() will return an already existing iris code from the test database and will set the mirror_attack_testcase_hack flag
+    // in this case we need to mirror the input so we can simulate the attack
+    // end result = mirrored code as input and the normal code existing in the test db
```

Comment:
ok for now, but as discussed, let's fix this in a follow-up and implement the proper `mirror()` function on the `IrisCode`